homeassistant/components/media_player/cambridgeaudio.py

```diff
@@ -0,0 +1,389 @@
+        return self._artist
+
+    @property
+    def media_album_name(self):
+        """Return album of current playing media."""
+        return self._album
+
+    @property
+    def media_track(self):
+        """Return track number of current playing media."""
+        return self._trackno
+
+    @property
+    def media_title(self):
+        """Return title of current playing media."""
+        return self._title
+
+    @property
+    def media_position(self):
+        """Return position of current playing media in seconds."""
+        if self._state is not STATE_PLAYING:
+            return None
+        return self._position
+
+    @property
+    def media_duration(self):
+        """Return the duration of current playing media in seconds."""
+        if self._state is not STATE_PLAYING:
+            return None
+        return self._duration
+
+    @property
+    def media_position_updated_at(self):
+        """When was the position of the current playing media valid.
+
+        Returns value from homeassistant.util.dt.utcnow().
+        """
+        if self._state in [STATE_PLAYING, STATE_PAUSED]:
+            return self._position_updated_at
+
+    @property
+    def volume_level(self):
+        """Current volume level (0..1)."""
+        return float(self._volume) / float(self._volume_max)
+
+    def clear_playlist(self):
+        """Clear the playlist."""
+        pass
+
+    def media_next_track(self):
+        """Skip to the next track, when in media player mode."""
+        if self._audio_source == "media player":
+            self._smdevice.trnsprt_next()
+            self.schedule_update_ha_state()
+
+    def media_previous_track(self):
+        """Skip to the previous track, when in media player mode."""
+        if self._audio_source == "media player":
+            self._smdevice.trnsprt_prev()
+            self.schedule_update_ha_state()
+
+    def media_pause(self):
+        """Pause playing the current media."""
+        # there is no "pause" with internet radio streams
+        if self._audio_source == "media player":
+            self._smdevice.trnsprt_pause()
+            self._state = STATE_PAUSED
+        elif self._audio_source == "internet radio":
+            self.media_stop()
+            self._state = STATE_IDLE
+
+    def media_seek(self, position):
+        """Jump to the specified percent position within the current track."""
+        abs_pos = round(self._to_seconds(self._duration) * (position/100), 2)
+        abs_pos_h = int(abs_pos / 3600)
+        abs_pos_m = int((abs_pos % 3600) / 60)
+        abs_pos_s = int((abs_pos % 3600) % 60)
+        abs_pos = '{:01d}:{:02d}:{:02d}'.format(abs_pos_h,
+                                                abs_pos_m,
+                                                abs_pos_s)
+        self._smdevice.trnsprt_seek(abs_pos)
+
+    def media_stop(self):
+        """Stop playing the current media."""
+        self._smdevice.trnsprt_stop()
+        self._state = STATE_IDLE
```

Comment:
You're doing optimistic updates but Home Assistant will call your update method right after. So are these state assignments necessary?

Reply:
See my last comment. I removed this now too.